src/main/python/org/broadinstitute/hellbender/tools/gcnv/gcnvkernel/models/model_ploidy.py

```diff
@@ -0,0 +1,364 @@
+
+from ..tasks.inference_task_base import HybridInferenceParameters
+from .fancy_model import GeneralizedContinuousModel
+from . import commons
+from .. import config, types
+from ..structs.interval import Interval
+from ..structs.metadata import IntervalListMetadata, SampleMetadataCollection
+
+_logger = logging.getLogger(__name__)
+
+
+class PloidyModelConfig:
+    """Germline contig ploidy model hyper-parameters."""
+
+    _contig_name_column = 'CONTIG_NAME'
+    _ploidy_prior_prefix = 'PLOIDY_PRIOR_'
+
+    def __init__(self,
+                 contig_ploidy_prior_map: Dict[str, np.ndarray] = None,
+                 mean_bias_sd: float = 1e-2,
+                 psi_j_scale: float = 1e-3,
+                 psi_s_scale: float = 1e-4,
+                 mapping_error_rate: float = 1e-2):
+        """Initializer.
+
+        Args:
+            contig_ploidy_prior_map: map from contigs to prior probabilities of each ploidy state
+            mean_bias_sd: standard deviation of mean contig-level coverage bias
+            psi_j_scale: typical scale of contig-specific unexplained variance
+            psi_s_scale: typical scale of sample-specific unexplained variance
+            mapping_error_rate: typical mapping error probability
+        """
+        assert contig_ploidy_prior_map is not None
+        self.mean_bias_sd = mean_bias_sd
+        self.psi_j_scale = psi_j_scale
+        self.psi_s_scale = psi_s_scale
+        self.mapping_error_rate = mapping_error_rate
+        self.contig_ploidy_prior_map, self.num_ploidy_states = self._get_validated_contig_ploidy_prior_map(
+            contig_ploidy_prior_map)
+        self.contig_set = set(contig_ploidy_prior_map.keys())
+        self.unordered_contig_list = list(self.contig_set)
+
+    @staticmethod
+    def _get_validated_contig_ploidy_prior_map(given_contig_ploidy_prior_map: Dict[str, np.ndarray],
+                                               min_prob: float = 0) -> Tuple[Dict[str, np.ndarray], int]:
+        given_contigs = set(given_contig_ploidy_prior_map.keys())
+        num_ploidy_states: int = 0
+        for contig in given_contigs:
+            num_ploidy_states = max(num_ploidy_states, given_contig_ploidy_prior_map[contig].size)
+        validated_contig_ploidy_prior_map: Dict[str, np.ndarray] = dict()
+        for contig in given_contigs:
+            padded_validated_prior = np.zeros((num_ploidy_states,), dtype=types.floatX) + min_prob
+            given_prior = given_contig_ploidy_prior_map[contig].flatten()
+            padded_validated_prior[:given_prior.size] = padded_validated_prior[:given_prior.size] + given_prior
+            padded_validated_prior = commons.get_normalized_prob_vector(padded_validated_prior, config.prob_sum_tol)
+            validated_contig_ploidy_prior_map[contig] = padded_validated_prior
+        return validated_contig_ploidy_prior_map, num_ploidy_states
+
+    @staticmethod
+    def get_contig_ploidy_prior_map_from_tsv_file(input_path: str,
```

Comment:
This seems like it should be in io_ploidy.py.

Reply:
Done.